plugins/module_utils/user.py

```diff
@@ -407,10 +407,11 @@ def user_mod(cursor, user, host, host_all, password, encrypted,
                         query_with_args = ("ALTER USER %s@%s IDENTIFIED WITH %s AS 0x" + generated_hash_string), (user, host, plugin)
                     else:
                         query_with_args = "ALTER USER %s@%s IDENTIFIED WITH %s BY %s", (user, host, plugin, plugin_auth_string)
-                else:
+                elif not module.check_mode:
                     query_with_args = "ALTER USER %s@%s IDENTIFIED WITH %s", (user, host, plugin)
 
-                cursor.execute(*query_with_args)
+                if not module.check_mode:
+                    cursor.execute(*query_with_args)
```

Comment:
```suggestion
                if not module.check_mode:
                    cursor.execute(*query_with_args)
```
Can we leave only this statement and remove all `not module.check_mode` occurrances above?